pkg/scheduler/plugins/v1alpha1/context.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	NotFound = "not found"
+)
+
+type ContextData interface{}
+type ContextKey string
+
+// Context provides a mechanism for plugins to store and retrieve arbitrary data.
+// ContextData stored by one plugin can be read, altered, or deleted by another plugin.
+// Context does not provide any data protection, as all plugins are assumed to be
+// trusted.
+type Context struct {
```

Comment:
This might confuse users with Go’s `Context.TODO()`. We might want to rename this into `unstructuredCache`